smithy-build/src/test/java/software/amazon/smithy/build/plugins/BuildInfoPluginTest.java

```diff
@@ -0,0 +1,51 @@
+package software.amazon.smithy.build.plugins;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.not;
+
+import java.net.URISyntaxException;
+import java.nio.file.Paths;
+import org.junit.jupiter.api.Test;
+import software.amazon.smithy.build.MockManifest;
+import software.amazon.smithy.build.PluginContext;
+import software.amazon.smithy.build.model.ProjectionConfig;
+import software.amazon.smithy.model.Model;
+import software.amazon.smithy.utils.ListUtils;
+
+public class BuildInfoPluginTest {
+
+    public static final String BUILD_INFO_PATH = "smithy-build-info.json";
+
+    public static final String MOCK_PROJECTION_NAME = "TEST";
+
+    public static final ProjectionConfig MOCK_PROJECTION_CONFIG = ProjectionConfig.builder().build();
+
+    public static final String SYNTHETIC_ENUM_TRAIT_ID = "smithy.synthetic#enum";
+
+    @Test
+    public void testFiltersSyntheticEnumTrait() throws URISyntaxException {
+
+        String testPath = "build-info-tests/example.smithy";
+
+
+        Model model = Model.assembler()
+                .addImport(getClass().getResource(testPath))
+                .assemble()
+                .unwrap();
+        MockManifest manifest = new MockManifest();
+        PluginContext context = PluginContext.builder()
+                .fileManifest(manifest)
+                .model(model)
+                .originalModel(model)
+                .projection(MOCK_PROJECTION_NAME, MOCK_PROJECTION_CONFIG)
```

Comment:
these aren't really mocks. I would just inline the values here instead of defining constants above.